IPC Cooling Unit tweaks

atlantiscze opened the pull request:
NOTE: Do not merge until @NullSnapshot approves the changes. The PR is only up for reviews and as a reference. Relevant forum topic: https://baystation12.net/forums/threads/suggestion-ipc-cooling-units.2347/
- Cooling Units no longer require a suit to be worn. Instead, they are worn on the back, similarly to how hardsuits are worn.
- Cooling Units now have overlays that update as the cell's charge level changes, similarly to how O2 tanks have indicator of remaining air. Should probably give you an early warning that you're running low on juice.
- Cooling Units now work even in hot environments. However, if the wearer lacks some form of insulation (a firesuit, for example), they will still heat up faster than it can cool them. As IPCs can't use atmospherics voidsuits which are commonly used in hot areas, this sounds as a good compromise to me.

// EDIT:
Also a side benefit of the above is that suit coolers don't need snowflake code for cryo chambers/mechas/etc.
- /proc/updateicon() refactored to global standard of update_icon().
- Some .loc = ... statements replaced with .forceMove()

atlantiscze committed:
IPC Cooling Unit tweaks
- Cooling Units no longer require a suit to be worn. Instead, they are worn on the back, similarly to how hardsuits are worn.
- Cooling Units now have overlays that update as the cell's charge level changes, similarly to how O2 tanks have indicator of remaining air. Should probably give you an early warning that you're running low on juice.
- Cooling Units now work even in hot environments. However, if the wearer lacks some form of insulation (a firesuit, for example), they will still heat up faster than it can cool them. As IPCs can't use atmospherics voidsuits which are commonly used in hot areas, this sounds as a good compromise to me.
- /proc/updateicon() refactored to global standard of update_icon().
- Some .loc = ... statements replaced with .forceMove()

NullSnapshot commented:
https://baystation12.net/forums/threads/suggestion-ipc-cooling-units.2347/#post-34709

Also do we have on back sprites?

atlantiscze commented:
They currently don't have on back sprites. My spriting skills are somewhere around the "Horrible" level, so there's zero chance i'd be making some. Though, how about re-using an existing sprite that is similar. The welding pack may look _relatively_ similar (at least until someone can make an actual dedicated sprite for it)

atlantiscze committed:
Tweaks
- Re-adds ability for the cooling unit to work when in a suit storage slot (assuming it fits into the suit)
- Adds stolen on-back sprites. Currently looks the same as welding kit, but still better than nothing.

NullSnapshot commented:
:+1: on species maintainer sign-off and code.
